misk-redis/src/main/kotlin/misk/redis/RedisModule.kt

```diff
@@ -6,34 +6,39 @@ import misk.ServiceModule
 import misk.inject.KAbstractModule
 import redis.clients.jedis.JedisPool
 import redis.clients.jedis.JedisPoolConfig
+import wisp.deployment.Deployment
 
 /**
  * Configures a JedisPool to connect to a Redis instance. The use of a JedisPool ensures thread safety.
  * See: https://github.com/xetorthio/jedis/wiki/Getting-started#using-jedis-in-a-multithreaded-environment
  */
 class RedisModule(
   private val redisConfig: RedisConfig,
-  private val jedisPoolConfig: JedisPoolConfig
+  private val jedisPoolConfig: JedisPoolConfig,
+  private val useSsl: Boolean = true,
 ) : KAbstractModule() {
   override fun configure() {
     bind<RedisConfig>().toInstance(redisConfig)
     install(ServiceModule<RedisService>())
   }
 
   @Provides @Singleton
-  internal fun provideRedisClient(config: RedisConfig): Redis {
-    // Get the first replication group, we only support 1 replication group per service
+  internal fun provideRedisClient(config: RedisConfig, deployment: Deployment): Redis {
+    // Get the first replication group, we only support 1 replication group per service.
     val replicationGroup = config[config.keys.first()]
-        ?: throw RuntimeException("At least 1 replication group must be specified")
+      ?: throw RuntimeException("At least 1 replication group must be specified")
 
     // Create our jedis pool
     val jedisPool = JedisPool(
-        jedisPoolConfig,
-        replicationGroup.writer_endpoint.hostname,
-        replicationGroup.writer_endpoint.port,
-        replicationGroup.timeout_ms,
-        replicationGroup.redis_auth_password,
-        true
+      jedisPoolConfig,
+      replicationGroup.writer_endpoint.hostname,
+      replicationGroup.writer_endpoint.port,
+      replicationGroup.timeout_ms,
+      replicationGroup.redis_auth_password.ifEmpty {
+        check(deployment.isFake) { "Redis auth password cannot be empty in a real environment!" }
+        null
+      },
```

Comment:
This is an important change. I could not for the life of me get the docker Redis image to accept an auth password at start-up without building a custom image and copying in a Redis config.

I don't think we're in the business of maintaining a docker image with a password just for the purpose of testing, so we'll allow an empty auth password in our configs **only** in fake environments.